### PR TITLE
OCPBUGS-14550: Use proxy with web socket connection and monitoring d…

### PR DESCRIPTION
**Pull request by Tafhim**
This PR is to demonstrate a change that enables new backplane to render a working monitoring dashboard and successful web socket connection.

**Note:** Blocking PR merge because this is intended to communicate the issue. Best design decisions are left to the console team.

### CODE REVIEW

**Comment by spadgett** [comment]
/hold

Does this only happen in a development environment? I would expect this to work in-cluster since we talk to the k8s service directly, which shouldn't need a proxy.

File: cmd/bridge/main.go [hunk 1]
```diff
@@ -766,6 +766,7 @@ func main() {
 		},
 		&http.Client{
 			Transport: &http.Transport{
+				Proxy:           http.ProxyFromEnvironment,
```
Comment (spadgett) [inline cmd/bridge/main.go:769]: I believe we should only set this if in off-cluster mode.

File: pkg/proxy/proxy.go [hunk 2]
```diff
@@ -218,6 +218,7 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	proxiedHeader.Add("Origin", "http://localhost")
 
 	dialer := &websocket.Dialer{
+		Proxy:           http.ProxyFromEnvironment,
```
[the remark this answers is not in the document]
Comment (spadgett) [reply pkg/proxy/proxy.go:221]: What if we add a parameter to `Config` indicating whether to use `ProxyFromEnvironment`?

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by yapei** [comment]
@jhadvig @spadgett could anyone help cherry-pick the fix to release-4.13 first?

**Pull request merged**